src/_pytest/_io/__init__.py

```diff
@@ -1,3 +1,39 @@
-# Reexport TerminalWriter from here instead of py, to make it easier to
-# extend or swap our own implementation in the future.
-from py.io import TerminalWriter as TerminalWriter  # noqa: F401
+from typing import List
+from typing import Sequence
+
+from py.io import TerminalWriter as BaseTerminalWriter  # noqa: F401
+
+
+class TerminalWriter(BaseTerminalWriter):
+    def _write_source(self, lines: List[str], indents: Sequence[str] = ()) -> None:
+        """Write lines of source code possibly highlighted.
+
+        Keeping this private for now because the API is clunky. We should discuss how
+        to evolve the terminal writer so we can have more precise color support, for example
+        being able to write part of a line in one color and the rest in another, and so on.
+        """
+        if indents and len(indents) != len(lines):
+            raise ValueError(
+                "indents size ({}) should have same size as lines ({})".format(
+                    len(indents), len(lines)
+                )
+            )
+        if not indents:
+            indents = [""] * len(lines)
+        source = "\n".join(lines)
+        new_lines = self._highlight(source).splitlines()
+        for indent, new_line in zip(indents, new_lines):
+            self.line(indent + new_line)
+
+    def _highlight(self, source):
+        """Highlight the given source code according to the "code_highlight" option"""
+        if not self.hasmarkup:
+            return source
+        try:
+            from pygments.formatters.terminal import TerminalFormatter
```

Comment:
Should we require `pygments` always? Doesn't seem to be that big of a dependency given the gains in readability.

Reply:
I'd suggest to add as extra in 5.4, call for feedback and turn regular based on the feedback in 5.5 or 6

Reply:
I did on my local copy, but after so thought I'm not sure it is a good idea: the extra would be named "pygments" or "coloring", and will then become part of our "API" because we will need to keep having that extra in there, even when it is no longer an extra, so users installing with `pip install pytest[pygments]` don't break. Also it won't save much typing.

Perhaps it is simple to just suggest users to install it manually, and let people comment if this should part of the regular dependencies on 5.5 or 6.0.

What do you say?